galley/pkg/config/processor/transforms/serviceentry/transformer.go

```diff
@@ -115,18 +115,26 @@ func (t *serviceEntryTransformer) Outputs() collection.Names {
 
 // Handle implements event.Transformer
 func (t *serviceEntryTransformer) Handle(e event.Event) {
+	output := t.Outputs()[0]
+
 	switch e.Kind {
 	case event.FullSync:
 		t.fullSyncCtr--
 		if t.fullSyncCtr == 0 {
-			t.dispatch(event.FullSyncFor(t.Outputs()[0]))
+			t.dispatch(event.FullSyncFor(output))
 		}
 		return
 
 	case event.Reset:
 		t.dispatch(event.Event{Kind: event.Reset})
 		return
 
+	// If any of the input resources are disabled, then SyntheticServiceEntries should also be disabled
+	case event.Disabled:
+		scope.Processing.Debugf("Received Disabled input event (%v), disabling output collection (%v)", e, output)
+		t.dispatch(event.DisabledFor(output))
```

Comment:
You probably want more than this. Specifically, once disabled, the transformer should not be sending out events.

Reply:
Got rid of this, obsolete.